api/s2n.h

```diff
@@ -424,6 +425,12 @@ extern int s2n_connection_is_ocsp_stapled(struct s2n_connection *conn);
 
 S2N_API
 extern struct s2n_cert_chain_and_key *s2n_connection_get_selected_cert(struct s2n_connection *conn);
+S2N_API
+extern int s2n_get_cert_chain_length(struct s2n_cert_chain_and_key *chain_and_key, uint32_t *cert_length);
+S2N_API
+extern int s2n_get_cert_from_cert_chain(struct s2n_cert_chain_and_key *chain_and_key, struct s2n_cert **out_cert, uint32_t cert_idx);
+S2N_API
+extern int s2n_get_cert_der(struct s2n_cert *cert, uint8_t *out_cert_der, uint32_t *cert_length);
```

Comment:
How does the caller know how much memory to allocate to receive the certificate?

Reply:
From offline discussion: 

To let the caller know how much memory to allocate to receive the certificate, we could:

- give the caller a way to know how much memory to allocate
- allocate the memory and own it
- allocate the memory and transfer ownership to the caller

Currently in `s2n_connection_get_client_cert_chain` we return the cert chain from wire in der format by allocating the memory and owning it. As such the memory allocated and returned will be valid only till the connection is alive.